test_util/compareRestart.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+   Copyright 2016 Statoil ASA.
+
+   This file is part of the Open Porous Media project (OPM).
+
+   OPM is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   OPM is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+   */
+
+#include <opm/output/eclipse/EclFilesComparator.hpp>
+
+#include <iostream>
+
+#include <ert/ecl/ecl_grid.h>
+
+static void printHelp() {
+    std::cout << "The program takes four arguments:\n"
+        << "1. .EGRID-file number 1\n" 
+        << "2. .EGRID-file number 2\n" 
+        << "2. .UNRST-file number 1\n" 
+        << "3. .UNRST-file number 2\n";
+}
+
+//------------------------------------------------//
+
+int main(int argc, char** argv) {
+    if (argc != 5) {
+        printHelp();
+        return 0;
+    }
+    try {
+        const char* gridFile1 = argv[1];
+        const char* gridFile2 = argv[2];
+        const char* unrstFile1 = argv[3];
+        const char* unrstFile2 = argv[4];
+        // Comparing grid sizes from .EGRID-files
+        {
+            ecl_grid_type* ecl_grid1 = ecl_grid_alloc(gridFile1);
+            ecl_grid_type* ecl_grid2 = ecl_grid_alloc(gridFile2); 
+            std::cout << "\nName of grid1: " << ecl_grid_get_name(ecl_grid1) << std::endl;
```

Comment:
There is a function `ecl_grid_equal (...)` which will compare all the vertices in the two grids,  this is a quite strict test which might fail when comparing an Eclipse reference grid and a grid from `flow` when `MINPV` or similar grid modifiers are active.